engine/src/main/java/com/datatorrent/stram/stream/BufferServerSubscriber.java

```diff
@@ -242,6 +243,21 @@ public long getByteCount(boolean reset)
     }
 
     @Override
+    public int size(final boolean dataTupleAware)
+    {
+      int size = size();
+      if (dataTupleAware) {
```

Comment:
@vrozov : Use case when you want to dynamically partition an operator based on queue size of backlog. For that you don't want to count control tuples, you want to calculate only data tuples..